ir/src/symbol_table.rs

```diff
@@ -185,8 +173,26 @@ impl SymbolTable {
 
     /// Consumes this symbol table and returns the information required for declaring constants,
     /// public inputs and periodic columns for the AIR.
-    pub(super) fn into_declarations(self) -> (Constants, PublicInputs, PeriodicColumns) {
-        (self.constants, self.public_inputs, self.periodic_columns)
+    pub(super) fn into_declarations(
+        self,
+    ) -> (Constants, PublicInputs, PeriodicColumns, TraceColumnGroups) {
+        let trace_column_groups = self
+            .identifiers
+            .iter()
+            .filter_map(|(name, ident_type)| {
+                if let IdentifierType::TraceColumnGroup(group) = ident_type {
+                    Some((name.clone(), group.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect();
```

Comment:
I think we can skip this for now and keep `into_declarations`as it was. I think anyone who wants a named trace column group for codegen may want the identifier info for the other columns too, so I think it makes sense to hold off on doing this until there's a use-case to define what's actually needed

Reply:
Makes sense. I was thinking the same. Removed.